pkg/common/client_state.go

```diff
@@ -59,6 +60,15 @@ func (i *ClientState) Read(context context.Context, cr *kc.KeycloakClient, realm
 		if err != nil {
 			return err
 		}
+
+		i.ScopeMappings, err = realmClient.ListScopeMappings(cr.Spec.Client.ID, i.Realm.Spec.Realm.Realm)
+		if err != nil {
+			return err
+		}
+	}
+
+	if i.ScopeMappings == nil {
+		i.ScopeMappings = &kc.MappingsRepresentation{}
```

Comment:
In most of the cases `ClientState` uses nils (that's why it uses pointers). Is there any reason to break this convention here and use an empty object?

Reply:
This was an oversight, I just deleted it, thanks.